### PR TITLE
implemented get devices

### PR DESCRIPTION
**Pull request by terloon**
## 📚 Purpose
Implements missing get devices ingest service call.

## 👌 Resolves:
Missing get devices call.

## 📦 Impacts:
* mds-ingest-service

### CODE REVIEW

File: packages/mds-ingest-service/repository/index.ts [hunk 1]
```diff
@@ -101,6 +104,16 @@ class IngestReadWriteRepository extends ReadWriteRepository {
     }
   }
 
+  public getDevices = async (ids: UUID[]): Promise<DeviceDomainModel[]> => {
+    try {
+      const connection = await this.connect('ro')
+      const entities = await connection.getRepository(DeviceEntity).find({ where: { device_id: In(ids) } })
```
Comment (drtyh2o) [inline packages/mds-ingest-service/repository/index.ts:110]: @nessur Would this benefit from your `= ANY([])` treatment?

**Review by nessur — APPROVED** [review]

File: packages/mds-ingest-service/service/validators.ts [hunk 2]
```diff
@@ -64,31 +64,27 @@ const telemetrySchema = Joi.object<TelemetryDomainModel>()
...
+          .valid(Joi.string().valid(...VEHICLE_EVENTS))
+          .required(),
+        vehicle_state: Joi.string().valid(...VEHICLE_STATES),
+        telemetry_timestamp: Joi.number().allow(null),
+        telemetry: telemetrySchema.allow(null),
+        trip_id: Joi.string().uuid().allow(null),
+        service_area_id: Joi.string().uuid().allow(null)
+      })
+      .unknown(false)
+  )
```
Comment (drtyh2o) [inline packages/mds-ingest-service/service/validators.ts:87]: Different version of `Prettier` installed? Why all the formatting changes?

**Review by drtyh2o — APPROVED** [review]
Looks good @terloon ... congrats on your first PR!

**Review by avatarneil — APPROVED** [review]

File: packages/mds-ingest-service/service/validators.ts (hunk 2 above)
Comment (terloon) [reply packages/mds-ingest-service/service/validators.ts:87]: Saw this as well. Are there standard formatting settings (tabs/spaces/etc) I should be abiding by? I can make sure to use it and recommit.

**Comment by avatarneil** [comment]
Congrats on the first PR :)

File: packages/mds-ingest-service/tests/index.spec.ts [hunk 3]
```diff
@@ -191,6 +191,23 @@ describe('Ingest Service Tests', () => {
     await IngestRepository.deleteAll()
   })
 
+  describe('getDevices', () => {
+    beforeEach(async () => {
+      const name = await IngestServiceClient.name()
+      await IngestRepository.createDevices([TEST_TNC_A, TEST_TNC_B])
+    })
+    describe('all_devices', () => {
```
Comment (BigIdeas) [inline packages/mds-ingest-service/tests/index.spec.ts:199]: It's a nit, but what about adding a test to call `getDevices` with both `DEVICE_UUID_A` and an invalid UUID to ensure it throws an exception? That way we'll know if someday the implementation changes to return a partial result instead. Just a thought.

**Review by BigIdeas — APPROVED** [review]

File: packages/mds-ingest-service/repository/index.ts (hunk 1 above)
Comment (nessur) [reply packages/mds-ingest-service/repository/index.ts]: thanks! TypeOrm has a wrapper method for that, even :)

File: packages/mds-ingest-service/tests/index.spec.ts (hunk 3 above)
Comment (terloon) [reply packages/mds-ingest-service/tests/index.spec.ts:198]: It's a good nit. Added two new tests for valid and invalid uuids.

File: packages/mds-ingest-service/service/validators.ts (hunk 2 above)
Comment (avatarneil) [reply packages/mds-ingest-service/service/validators.ts:87]: We have rules defined in `.prettierrc.json` and `.eslintrc.json` which define our formatting standards. If you use VSCode and install the extensions recommended by the `.vscode/extensions.json` file in this repo, you ought to auto-magically have all the formatting rules applied. Also, I think that they're typically applied as part of a pre-commit linting hook.

**Pull request merged**